assembly/std/solidity_compat/misc.ts

```diff
@@ -10,3 +15,27 @@ import { wrapStaticArray } from '@massalabs/as-types';
 export function bytes32ToU256(a: StaticArray<u8>): u256 {
   return u256.fromUint8ArrayBE(wrapStaticArray(a));
 }
+
+// implementation Solidity selfdestruct:
+// https://www.infuy.com/blog/using-self-destruct-in-solidity-contracts/
+// Note: this function will try to delete all Storage keys. Devs might want to clean things up manually
+//       before using this function.
+export function selfDestruct(transferToAddr: Address): void {
+  // 1- empty the SC
+  let emptySc = new StaticArray<u8>(0);
+  env.setBytecode(emptySc);
+
+  // 2- delete everything in Storage
+  let keys = env.getKeys(new StaticArray<u8>(0));
```

Comment:
I think you want to use `getKeys()` in storage.ts, not `env.getKeys`
`env.getKeys`return one single buffer altough `getKeys()` while "deserialize" the buffer into an array of keys